lib/internal/http2/util.js

```diff
@@ -415,16 +417,18 @@ function mapToHeaders(map,
       if (isArray) {
         for (var k = 0; k < value.length; k++) {
           val = String(value[k]);
-          ret.push([key, val]);
+          ret += `${key}\0${val}\0`;
```

Comment:
This presupposes that neither the key nor the val contain null characters. That may not be a safe assumption to make